src/components/invoice-form.tsx

```diff
@@ -38,348 +53,821 @@ interface InvoiceFormProps {
+      <div className="flex justify-between items-center w-full">
+        <span>
+          {detail.paymentDetails.accountName || "Account"} •••• {lastFourDigits}
+        </span>
+        {payer && <PaymentDetailsStatus status={payer.status} />}
+      </div>
+    </SelectItem>
+  );
+};
+
+// Payment Details Loading State
+const PaymentDetailsLoading = () => (
+  <div className="text-sm text-gray-500">Checking client details...</div>
+);
+
+// Payment Details Error States
+const PaymentDetailsError = ({ message }: { message: string }) => (
+  <p className="text-sm text-gray-500">{message}</p>
+);
+
+// Payment Details Empty State
+const PaymentDetailsEmpty = ({ onAdd }: { onAdd: () => void }) => (
+  <div className="space-y-4">
+    <p className="text-sm text-gray-500">
+      No bank accounts found for this client
+    </p>
+    <Button type="button" variant="outline" onClick={onAdd} className="w-full">
+      Add Bank Account
+    </Button>
+  </div>
+);
+
+// Payment Details Select Component
+const PaymentDetailsSelect = ({
+  details,
+  clientEmail,
+  onSelect,
+  defaultValue,
+}: {
+  details: {
+    paymentDetails: PaymentDetails;
+    paymentDetailsPayers: (User & PaymentDetailsPayers)[];
+  }[];
+  clientEmail: string;
+  onSelect: (value: string) => void;
+  defaultValue?: string;
+}) => (
+  <div className="space-y-2">
+    <Select onValueChange={onSelect} defaultValue={defaultValue}>
+      <SelectTrigger>
+        <SelectValue placeholder="Select payment details" />
+      </SelectTrigger>
+      <SelectContent>
+        {details.map((detail) => (
+          <PaymentDetailsSelectItem
+            key={detail.paymentDetails.id}
+            detail={detail}
+            clientEmail={clientEmail}
+          />
+        ))}
+      </SelectContent>
+    </Select>
+  </div>
+);
+
+// Payment Details Section Component
+const PaymentDetailsSection = ({
+  clientEmail,
+  isLoadingUser,
+  clientUserData,
+  linkedPaymentDetails,
+  onAddBankAccount,
+  onSelectPaymentDetails,
+  selectedPaymentDetailsId,
+}: {
+  clientEmail: string;
+  isLoadingUser: boolean;
+  clientUserData: User | null | undefined;
+  linkedPaymentDetails:
+    | {
+        paymentDetails: PaymentDetails;
+        paymentDetailsPayers: (User & PaymentDetailsPayers)[];
+      }[]
+    | undefined;
+  onAddBankAccount: () => void;
+  onSelectPaymentDetails: (value: string) => void;
+  selectedPaymentDetailsId?: string;
+}) => {
+  if (!clientEmail) {
+    return (
+      <PaymentDetailsError message="Enter client email to view payment details" />
+    );
+  }
+
+  if (isLoadingUser) {
+    return <PaymentDetailsLoading />;
+  }
+
+  if (!clientUserData) {
+    return (
+      <PaymentDetailsError message="Client not compliant. Please use a verified client" />
+    );
+  }
+
+  if (!clientUserData.isCompliant) {
+    return (
+      <PaymentDetailsError message="Client is not valid for Crypto to Fiat" />
+    );
+  }
```

Comment:
Ambiguous error messages. Possible mismatch.

The message "Client not compliant..." seems to match the `if (!clientData.isCompliant)` better than the `(!clientUserData)`.

Please clarify the error and the mitigation, but keep it as short as possible.

```suggestion
  if (!clientUserData) {
    return (
      <PaymentDetailsError message="Client has not completed KYC. Please use a verified client." />
    );
  }

  if (!clientUserData.isCompliant) {
    return (
      <PaymentDetailsError message="Client's KYC application was rejected. Please use a verified client." />
    );
  }
```